app/templates/signup.hbs

```diff
@@ -1,27 +1,38 @@
 <form class="form">
   <h2>{{this.title}}</h2>
   {{#each this.fields as |field|}}
+  {{#let this.formData as | myFormData |}}
     <FormInput
       @id={{field.id}}
       @for={{field.for}}
       @label={{field.label}}
       @type={{field.type}}
-      @value=''
+      @value={{get this.formData field.id}}
       @errorMessage={{field.errorMessage}}
       @required={{field.required}}
       @showError={{field.showError}}
       @onChange={{this.handleFieldChange}}
-      @validator={{field.validator}}
+      {{!-- @validator={{field.validator}} --}}
     />
+    {{/let}}
   {{/each}}
-  <button
-    class={{if this.isSubmitDisabled 'submitButton disabledButton' 'submitButton'}}
-    type="submit"
-    disabled={{this.isSubmitDisabled}}
-    {{on 'click' (fn this.handleSubmit)}}
-  >
-    Submit
-  </button>
+  {{#if this.isSubmitDisabled }}
+    <button
+      class='submitButton disabledButton'
+      type="submit"
+      disabled={{true}}
+    >
+      Submit
+    </button>
+    {{else}}
+      <button
+      class='submitButton'
+      type="submit"
+      {{on 'click' (fn this.handleSubmit)}}
+      >
```

Comment:
Please tell me why fn helper is used here!

I think fn helper should not be here.

Reply:
Can you please provide some reference?